sale_order_confirm_partial/tests/test_sale_order_confirm_partial.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import exceptions
+from odoo.tests import TransactionCase, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestSaleOrderConfirmPartial(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env["ir.config_parameter"].set_param(
+            "sale_order_confirm_partial.enabled", True
+        )
+        cls.sale_order = cls.env["sale.order"].create(
+            {
+                "name": "SO-TEST",
+                "partner_id": cls.env.ref("base.res_partner_1").id,
+                "pricelist_id": cls.env.ref("product.list0").id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.env.ref("product.product_product_6").id,
+                            "product_uom_qty": 10,
+                            "price_unit": 100,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.env.ref("product.product_product_7").id,
+                            "product_uom_qty": 7,
+                            "price_unit": 200,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    def test_confirmation_with_save_unconfirmed_all(self):
+        self.env["ir.config_parameter"].set_param(
+            "sale_order_confirm_partial.save_unconfirmed", True
+        )
+        # Confirm SO partially
+        self.env["sale.order.confirm.partial"].create(
+            {"sale_order_id": self.sale_order.id, "mode": "all"}
+        ).action_confirm()
+
+        # Check SO state
+        self.assertEqual(self.sale_order.state, "sale", "Sale order is not confirmed")
+
+        # Since we have confirmed all of the lines, there is no need to save
+        # the unconfirmed part of SO
+        unconfirmed_so = self.env["sale.order"].search([("name", "like", "SO-TEST-U")])
+        self.assertFalse(
+            unconfirmed_so,
+            "Unconfirmed part of SO has been created when all of lines were confirmed.",
+        )
+
+        # Check original order lines quantities
+        # (they should stay the same)
+        self.assertEqual(
+            self.sale_order.order_line[0].product_uom_qty,
```

Comment:
This is what I was talking about in the previous comment. You need to use direct reference. Eg
```
self.so_line_1.product_uom_qty,
```